apps/api/src/app/workflows-v2/usecases/upsert-workflow/upsert-workflow.dto.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  ArrayMaxSize,
+  IsArray,
+  IsBoolean,
+  IsEnum,
+  IsNotEmpty,
+  IsOptional,
+  IsString,
+  Length,
+  ValidateNested,
+} from 'class-validator';
+import { Type } from 'class-transformer';
+import { MAX_NAME_LENGTH } from '@novu/application-generic';
+import { WorkflowCreationSourceEnum } from '@novu/shared';
+import { UpsertStepData } from './upsert-step.data';
+import { PreferencesRequestUpsertData } from './preferences-request-upsert.data';
+
+export class UpsertWorkflowDto {
```

Comment:
**General note:**
now that we have a separate class for the API we need to rethink how we maintain the DTO's in shared, because not we could have misalignments

Reply:
Correct, the DTO needs to be an interface in `@novu/shared`. There is another draft PR where I replace all classes in DTOs with interfaces to make strict TS happy.

Reply:
Don't we also need an optional `_id` field for the Mongo Object ID?

Reply:
Hold your horses people, interfaces and classes are the same for that reason 
![image](https://github.com/user-attachments/assets/5f2e50ef-89d0-4f16-a301-27a29564951f)
The solution here, and it will take probably a month more is to move all DTO's into the API and have our open api generator generate them as a part of the SDK, this is the only thing that will achieve the correct seperation here, that's next on the list for the SDK work, I would appreciate if we wait for that